### ISSUE TITLE
Unnecessary parenthesis detection on GCC9

### ISSUE THREAD

**Issue by eldruin** [ukoethe/vigra#479]
Our GCC9 build reports an unnecessary parenthesis error in [this code](https://github.com/ukoethe/vigra/blob/cc5effe52402c8591c868ee608a104caa9924ad9/include/vigra/static_assert.hxx#L87):
```
template< typename Predicate >
failure ************ (Predicate::************ 
      assertImpl( void (*)(Predicate), typename Predicate::not_type )
    );
```

The exact message is:

    include/vigra/static_assert.hxx:87 unnecessary parentheses in declaration of ‘assertImpl’ [-Werror=parentheses]

Would it be possible to change those messages?

**Comment by hmeine** [comment]
I could finally reproduce it in a docker with gcc9 and a simple `make check` in a fresh vigra checkout, many dependencies missing:
```
In file included from /vigra/include/vigra/tinyvector.hxx:57,
                 from /vigra/include/vigra/voxelneighborhood.hxx:39,
                 from /vigra/test/voxelneighborhood/test.cxx:42:
/vigra/include/vigra/static_assert.hxx: At global scope:
/vigra/include/vigra/static_assert.hxx:87:22: warning: unnecessary parentheses in declaration of 'assertImpl' [-Wparentheses]
   87 | failure ************ (Predicate::************
      |                      ^
```

**Comment by hmeine** [comment]
Fixed in aaaf44d240aebd0eb3539b6e105a19a984000d81
